### PR TITLE
RFR - Tasks in action chain inherit notify from chain

### PR DESCRIPTION
**Pull request by lakshmi-kannan**
Without skip_notify:

```
~/st2 git:chain_inherit_notifications ❯❯❯ st2 run examples.echochain                                                  ⏎ ✭ ✱ ◼
..
id: 555223ff6d0aff6d610ca22f
action.ref: examples.echochain
status: succeeded
start_timestamp: 2015-05-12T16:02:07.610206Z
end_timestamp: 2015-05-12T16:02:10.853135Z
+--------------------------+-----------+------+------------+----------------------+
| id                       | status    | task | action     | start_timestamp      |
+--------------------------+-----------+------+------------+----------------------+
| 555223ff6d0aff6d6779609b | succeeded | c1   | core.local | Tue, 12 May 2015     |
|                          |           |      |            | 16:02:07 UTC         |
| 555224006d0aff6d6779609f | succeeded | c2   | core.local | Tue, 12 May 2015     |
|                          |           |      |            | 16:02:08 UTC         |
| 555224016d0aff6d677960a3 | succeeded | c3   | core.local | Tue, 12 May 2015     |
|                          |           |      |            | 16:02:09 UTC         |
+--------------------------+-----------+------+------------+----------------------+
~/st2 git:chain_inherit_notifications ❯❯❯ st2 execution list                                                            ✭ ✱ ◼
+----------------------------+------------------+--------------+-----------+------------------+------------------+
| id                         | action.ref       | context.user | status    | start_timestamp  | end_timestamp    |
+----------------------------+------------------+--------------+-----------+------------------+------------------+
| + 555223ff6d0aff6d610ca22f | examples.echocha | stanley      | succeeded | Tue, 12 May 2015 | Tue, 12 May 2015 |
|                            | in               |              |           | 16:02:07 UTC     | 16:02:10 UTC     |
|   555223ff6d0aff6d6d708577 | slack.post_messa | stanley      | succeeded | Tue, 12 May 2015 | Tue, 12 May 2015 |
|                            | ge               |              |           | 16:02:07 UTC     | 16:02:13 UTC     |
|   555224006d0aff6d6d70857b | slack.post_messa | stanley      | succeeded | Tue, 12 May 2015 | Tue, 12 May 2015 |
|                            | ge               |              |           | 16:02:08 UTC     | 16:02:14 UTC     |
|   555224016d0aff6d6d70857f | slack.post_messa | stanley      | succeeded | Tue, 12 May 2015 | Tue, 12 May 2015 |
|                            | ge               |              |           | 16:02:09 UTC     | 16:02:15 UTC     |
|   555224026d0aff6d6d708583 | slack.post_messa | stanley      | running   | Tue, 12 May 2015 |                  |
|                            | ge               |              |           | 16:02:10 UTC     |                  |
+----------------------------+------------------+--------------+-----------+------------------+------------------+
~/st2 git:chain_inherit_notifications ❯❯❯ st2 execution list                                                            ✭ ✱ ◼
+----------------------------+------------------+--------------+-----------+------------------+------------------+
| id                         | action.ref       | context.user | status    | start_timestamp  | end_timestamp    |
+----------------------------+------------------+--------------+-----------+------------------+------------------+
| + 555223ff6d0aff6d610ca22f | examples.echocha | stanley      | succeeded | Tue, 12 May 2015 | Tue, 12 May 2015 |
|                            | in               |              |           | 16:02:07 UTC     | 16:02:10 UTC     |
|   555223ff6d0aff6d6d708577 | slack.post_messa | stanley      | succeeded | Tue, 12 May 2015 | Tue, 12 May 2015 |
|                            | ge               |              |           | 16:02:07 UTC     | 16:02:13 UTC     |
|   555224006d0aff6d6d70857b | slack.post_messa | stanley      | succeeded | Tue, 12 May 2015 | Tue, 12 May 2015 |
|                            | ge               |              |           | 16:02:08 UTC     | 16:02:14 UTC     |
|   555224016d0aff6d6d70857f | slack.post_messa | stanley      | succeeded | Tue, 12 May 2015 | Tue, 12 May 2015 |
|                            | ge               |              |           | 16:02:09 UTC     | 16:02:15 UTC     |
|   555224026d0aff6d6d708583 | slack.post_messa | stanley      | succeeded | Tue, 12 May 2015 | Tue, 12 May 2015 |
|                            | ge               |              |           | 16:02:10 UTC     | 16:02:16 UTC     |
+----------------------------+------------------+--------------+-----------+------------------+------------------+
~/st2 git:chain_inherit_notifications ❯❯❯
```

With skip_notify:

```
~/st2 git:chain_inherit_notifications ❯❯❯ ST2_CONF=~/st2.dev.conf tools/launchdev.sh stop                               ✭ ✱ ◼
Killing existing st2 screen sessions...
~/st2 git:chain_inherit_notifications ❯❯❯ ST2_CONF=~/st2.dev.conf tools/launchdev.sh startclean                         ✭ ✱ ◼
MongoDB shell version: 2.6.3
connecting to: st2
[object Object]
Starting all st2 servers...
Changing working directory to /home/lakshmi/st2/tools/.....
Using st2 config file: /home/lakshmi/st2.dev.conf
Using content packs base dir: /opt/stackstorm/packs
There is a screen on:
	28002.st2-actionrunner	(05/12/2015 04:01:55 PM)	(Detached)
1 Socket in /var/run/screen/S-lakshmi.

Killing existing st2 screen sessions...
Starting screen session st2-api...
Starting screen session st2-actionrunner...
    starting runner  1 ...
Starting screen session st2-sensorcontainer
Starting screen session st2-rulesengine...
Starting screen session st2-resultstracker...
Starting screen session st2-notifier...

Registering sensors, actions, rules and aliases...
2015-05-12 16:03:17,643 INFO [-] Connecting to database "st2" @ "0.0.0.0:27017" as user "None".
2015-05-12 16:03:17,914 INFO [-] =========================================================
2015-05-12 16:03:17,919 INFO [-] ############## Registering sensors ######################
2015-05-12 16:03:17,919 INFO [-] =========================================================
2015-05-12 16:03:18,408 INFO [-] Registered 4 sensors.
2015-05-12 16:03:18,408 INFO [-] =========================================================
2015-05-12 16:03:18,408 INFO [-] ############## Registering actions ######################
2015-05-12 16:03:18,408 INFO [-] =========================================================
2015-05-12 16:03:19,112 INFO [-] Registered 55 actions.
2015-05-12 16:03:19,112 INFO [-] =========================================================
2015-05-12 16:03:19,112 INFO [-] ############## Registering rules ########################
2015-05-12 16:03:19,112 INFO [-] =========================================================
2015-05-12 16:03:19,166 INFO [-] Registered 4 rules.
2015-05-12 16:03:19,166 INFO [-] =========================================================
2015-05-12 16:03:19,166 INFO [-] ############## Registering aliases ######################
2015-05-12 16:03:19,167 INFO [-] =========================================================
2015-05-12 16:03:19,171 INFO [-] ActionAlias /opt/stackstorm/packs/deploy/aliases/pack_delete.yaml not found. Creating new one.
2015-05-12 16:03:19,179 INFO [-] ActionAlias /opt/stackstorm/packs/deploy/aliases/pack_deploy.yaml not found. Creating new one.
2015-05-12 16:03:19,188 INFO [-] ActionAlias /opt/stackstorm/packs/deploy/aliases/pack_info.yaml not found. Creating new one.
2015-05-12 16:03:19,189 INFO [-] Registered 3 aliases.
There are screens on:
	28624.st2-resultstracker	(05/12/2015 04:03:17 PM)	(Detached)
	28626.st2-notifier	(05/12/2015 04:03:17 PM)	(Detached)
	28620.st2-rulesengine	(05/12/2015 04:03:17 PM)	(Detached)
	28617.st2-sensorcontainer	(05/12/2015 04:03:17 PM)	(Detached)
	28611.st2-actionrunner	(05/12/2015 04:03:17 PM)	(Detached)
	28608.st2-api	(05/12/2015 04:03:17 PM)	(Detached)
6 Sockets in /var/run/screen/S-lakshmi.

~/st2 git:chain_inherit_notifications ❯❯❯ st2 run examples.echochain                                                  ⏎ ✭ ✱ ◼
..
id: 5552244a6d0aff6fc280197b
action.ref: examples.echochain
status: succeeded
start_timestamp: 2015-05-12T16:03:22.145231Z
end_timestamp: 2015-05-12T16:03:25.378969Z
+--------------------------+-----------+------+------------+----------------------+
| id                       | status    | task | action     | start_timestamp      |
+--------------------------+-----------+------+------------+----------------------+
| 5552244a6d0aff6fc7c9e9a5 | succeeded | c1   | core.local | Tue, 12 May 2015     |
|                          |           |      |            | 16:03:22 UTC         |
| 5552244b6d0aff6fc7c9e9a9 | succeeded | c2   | core.local | Tue, 12 May 2015     |
|                          |           |      |            | 16:03:23 UTC         |
| 5552244c6d0aff6fc7c9e9ac | succeeded | c3   | core.local | Tue, 12 May 2015     |
|                          |           |      |            | 16:03:24 UTC         |
+--------------------------+-----------+------+------------+----------------------+
~/st2 git:chain_inherit_notifications ❯❯❯ st2 execution list                                                            ✭ ✱ ◼
+----------------------------+------------------+--------------+-----------+------------------+------------------+
| id                         | action.ref       | context.user | status    | start_timestamp  | end_timestamp    |
+----------------------------+------------------+--------------+-----------+------------------+------------------+
| + 5552244a6d0aff6fc280197b | examples.echocha | stanley      | succeeded | Tue, 12 May 2015 | Tue, 12 May 2015 |
|                            | in               |              |           | 16:03:22 UTC     | 16:03:25 UTC     |
|   5552244a6d0aff6fce293025 | slack.post_messa | stanley      | succeeded | Tue, 12 May 2015 | Tue, 12 May 2015 |
|                            | ge               |              |           | 16:03:22 UTC     | 16:03:28 UTC     |
|   5552244c6d0aff6fce29302a | slack.post_messa | stanley      | running   | Tue, 12 May 2015 |                  |
|                            | ge               |              |           | 16:03:24 UTC     |                  |
|   5552244d6d0aff6fce29302e | slack.post_messa | stanley      | running   | Tue, 12 May 2015 |                  |
|                            | ge               |              |           | 16:03:25 UTC     |                  |
+----------------------------+------------------+--------------+-----------+------------------+------------------+
~/st2 git:chain_inherit_notifications ❯❯❯ st2 execution list                                                            ✭ ✱ ◼
+----------------------------+------------------+--------------+-----------+------------------+------------------+
| id                         | action.ref       | context.user | status    | start_timestamp  | end_timestamp    |
+----------------------------+------------------+--------------+-----------+------------------+------------------+
| + 5552244a6d0aff6fc280197b | examples.echocha | stanley      | succeeded | Tue, 12 May 2015 | Tue, 12 May 2015 |
|                            | in               |              |           | 16:03:22 UTC     | 16:03:25 UTC     |
|   5552244a6d0aff6fce293025 | slack.post_messa | stanley      | succeeded | Tue, 12 May 2015 | Tue, 12 May 2015 |
|                            | ge               |              |           | 16:03:22 UTC     | 16:03:28 UTC     |
|   5552244c6d0aff6fce29302a | slack.post_messa | stanley      | succeeded | Tue, 12 May 2015 | Tue, 12 May 2015 |
|                            | ge               |              |           | 16:03:24 UTC     | 16:03:29 UTC     |
|   5552244d6d0aff6fce29302e | slack.post_messa | stanley      | succeeded | Tue, 12 May 2015 | Tue, 12 May 2015 |
|                            | ge               |              |           | 16:03:25 UTC     | 16:03:31 UTC     |
+----------------------------+------------------+--------------+-----------+------------------+------------------+
~/st2 git:chain_inherit_notifications ❯❯❯
```

### CODE REVIEW

File: contrib/examples/actions/echochain.meta.yaml [hunk 1]
```diff
@@ -10,4 +10,11 @@ runner_type: "action-chain"
 entry_point: "chains/echochain.yaml"
 
 enabled: true
-parameters: {}
+parameters:
+  skip_notify:
```
Comment (Kami) [inline contrib/examples/actions/echochain.meta.yaml]: Wait, confused, shouldn't this be like:

```yaml
skip_notify:
  - c2
```

Or am I missing something?

Comment (Kami) [inline contrib/examples/actions/echochain.meta.yaml]: Edit: I see, it's a string.

Any reason why it's a string and not a list (array)?

Comment (lakshmi-kannan) [inline contrib/examples/actions/echochain.meta.yaml]: It's a runner parameter. So with CLI you have to use it as a string. It's unfortunate.

Comment (Kami) [inline contrib/examples/actions/echochain.meta.yaml]: Still confused - what has CLI to do with it?

We have a native list type and we should use that? And if you "comma delimited string" syntax in the CLI and the type of the param is an array, CLI will automatically parse it and send it as a list so there should be no issues.

Comment (lakshmi-kannan) [inline contrib/examples/actions/echochain.meta.yaml]: Does it? I probably missed this. It used to be that CLI didn't parse a comma separated list as an array.

Comment (Kami) [inline contrib/examples/actions/echochain.meta.yaml]: It did this as long as I can remember :P

In any case, as I noted above - CLI is just one type of our API consumer (others are programming language clients, webui, etc.). We should not bias a general purpose API towards a single consumer.

File: st2actions/st2actions/runners/actionchainrunner.py [hunk 2]
```diff
@@ -325,8 +335,15 @@ def _run_action(self, action_node, parent_execution_id, params, wait_for_complet
         liveaction = LiveActionDB(action=action_node.ref)
         liveaction.parameters = action_param_utils.cast_params(action_ref=action_node.ref,
                                                                params=params)
-        if action_node.notify:
-            liveaction.notify = NotificationsHelper.to_model(action_node.notify)
+
+        # Setup notify for task in chain.
+        if action_node.name not in self._skip_notify_tasks:
```
Comment (Kami) [inline st2actions/st2actions/runners/actionchainrunner.py]: Might be a bit easier to read if it's refactored in a separate method - e.g. `_should_notify_for(task_name)`.

**Comment by Kami** [comment]
Besides the things I have mentioned, this approach looks good to me.

File: st2actions/st2actions/runners/actionchainrunner.py (hunk 2 above)
Comment (lakshmi-kannan) [inline st2actions/st2actions/runners/actionchainrunner.py]: Got it as part of ecbcf56

File: contrib/examples/actions/echochain.meta.yaml (hunk 1 above)
Comment (lakshmi-kannan) [inline contrib/examples/actions/echochain.meta.yaml]: Agreed, see ecbcf56.

**Pull request merged**